### PR TITLE
feat: adding Postgres memory for message persistent

### PR DESCRIPTION
**Pull request by PepperYan**
Adding PostgresChatHistory for buffer memory enhancement(messages could be storage in db). 

but I found it frontend doesnt support the link from Memory to ChatHistory. Might be some additional works to do.(the output of /all works fine)

If there is any questions pls let me know.

### CODE REVIEW

**Comment by PepperYan** [comment]
session_id in pg should be pass and replace by tweak, in langflow scenario it might be client_id

**Comment by ogabrielluiz** [comment]
Hey, @PepperYan. Thanks for this PR. We need to do a couple of things to make this work.

First, in the `MemoryCreator` the  `get_signature` method should look something like this:
```python
def get_signature(self, name: str) -> Optional[Dict]:
        """Get the signature of a memory."""
        try:
            if name in get_custom_nodes(self.type_name).keys():
                return get_custom_nodes(self.type_name)[name]
            return build_template_from_class(name, memory_type_to_cls_dict)
        except ValueError as exc:
            raise ValueError("Memory not found") from exc
        except AttributeError as exc:
            logger.error(f"Memory {name} not loaded: {exc}")
            return None
```

Now, in the  `MemoryFrontendNode` `add_extra_fields` method, two things (I think) should be changed:
You should add this:
```python
def add_extra_fields(self) -> None:
        from langflow.custom.customs import get_custom_nodes

        # add return_messages field
        if self.name not in get_custom_nodes("memories").keys():
```
And the other is adding a `chat_memory` field to the Memories, so that we can connect the ChatMessageHistory Components to it.

**Comment by PepperYan** [comment]
thanks for pointing out the problems of get_signatrue,  I have make a commit to fix these

File: src/backend/langflow/template/frontend_node/memories.py [hunk 1]
```diff
@@ -2,11 +2,17 @@
 
 from langflow.template.field.base import TemplateField
 from langflow.template.frontend_node.base import FrontendNode
+from langflow.template.template.base import Template
 
 
 class MemoryFrontendNode(FrontendNode):
     #! Needs testing
     def add_extra_fields(self) -> None:
+        # chat history should have another way to add common field?
```
Comment (ogabrielluiz) [inline src/backend/langflow/template/frontend_node/memories.py:11]: I agree. With your PR we might need to consider a new category. This solution will do for now.

**Comment by ogabrielluiz** [comment]
So, have you managed to build a working example?

**Comment by PepperYan** [comment]
> 

psycopg-c might not be necessary. But langchain PostgresChatMessageHistory require psycopg, If I do not add dependencies in poetry, it might be some error.
`class PostgresChatMessageHistory(BaseChatMessageHistory):
    """Chat message history stored in a Postgres database."""

    def __init__(
        self,
        session_id: str,
        connection_string: str = DEFAULT_CONNECTION_STRING,
        table_name: str = "message_store",
    ):
        import psycopg
        from psycopg.rows import dict_row

        try:
            self.connection = psycopg.connect(connection_string)
            self.cursor = self.connection.cursor(row_factory=dict_row)
        except psycopg.OperationalError as error:
            logger.error(error)

        self.session_id = session_id
        self.table_name = table_name

        self._create_table_if_not_exists()`

**Comment by PepperYan** [comment]
@ogabrielluiz 
I just make a example for my PR, pls see

https://github.com/PepperYan/langflow_examples/commit/9efa71f58b70b7b151a14d7c24b99c2899bf7e7e

**Comment by PepperYan** [comment]
> This looks good to me. I had to install PostgreSQL which is not ideal (because not everyone will use that component) but we can think of something to get around that.
> 
> Oh... don't forget to run `poetry lock` and commit that so that we can install everything. Is `psycopg-c` really needed?
> 
> So, have you managed to build a working example?

I thought message persistences would be a lovely feature. But indeed we should find a elegant way to prevent intrusion from 3rd party library

**Comment by ogabrielluiz** [comment]
> @ogabrielluiz I just make a example for my PR, pls see
> 
> [PepperYan/langflow_examples@9efa71f](https://github.com/PepperYan/langflow_examples/commit/9efa71f58b70b7b151a14d7c24b99c2899bf7e7e)

I'll approve it as soon as we get this PR out to the public. I'll merge it into our next minor release.

**Pull request merged**